src/ShaderGen/LanguageBackend.cs

```diff
@@ -359,5 +360,40 @@ internal virtual string CorrectCastExpression(string type, string expression)
         {
             return $"({type}) {expression}";
         }
+
+        protected virtual ShaderMethodVisitor VisitShaderMethod(Compilation compilation,string setName,ShaderFunction func)
+        {
+            return new ShaderMethodVisitor(Compilation, setName,func, this);
```

Comment:
This function takes a `compilation` parameter but then uses the `Compilation` property. Given how this function is used and overridden currently, I think we should just remove it. It's not doing anything interesting and is doing more harm than good, IMO.

Reply:
nvm -- as you mentioned, derived classes are using different ShaderMethodVisitor subclasses. Still, this should either use the `compilation` parameter passed in, or all of the overloads should just use the `Compilation` field.